test/test_OCImock.py

```diff
@@ -12,38 +9,34 @@ def setup_class(cls):
     def teardown_class(cls):
         pass
 
+    # Upload a file
+    @pytest.mark.skipif(env_vars_absent() == False, reason="Environment variable(s) should not be defined")
     def test_upload(self, file, powerloader_binary):
         upload_path = generate_unique_file(file)
         hash_before_upload = calculate_sha256(upload_path)
-        tag, name_on_server = upload_oci(upload_path, powerloader_binary, file["oci_upload_location"])
-
-    def test_download_permanent(self, file, powerloader_binary, checksums):
-        Path(get_oci_path(file=file)[1]).unlink(missing_ok=True)
-        newpath, tmp_yaml = generate_oci_download_yml(file)
-        download_oci_file(powerloader_binary, tmp_yaml, file)
-        assert checksums[file["name_on_server"]] == calculate_sha256(newpath)
-
-    def set_username(self):
-        username = ""
-        if self.username_exists():
-            username = os.environ.get("GHA_USER")
-        else:
-            username = "mamba-org"              # GHA_PAT is only available on the main branch
-            os.environ["GHA_USER"] = username   # GHA_USER must also be set
-        return username
+        tag, name_on_server = upload_oci(upload_path, powerloader_binary, file["oci_mock_server"], plain_http=True)
 
     # Download a file that's always there...
-    @pytest.mark.skipif(os.environ.get("GHA_PAT") is None
-                        or os.environ.get("GHA_PAT") == "",
-                        reason="Environment variable(s) not defined")
+    @pytest.mark.skipif(env_vars_absent() == False, reason="Environment variable(s) should not be defined")
+    def test_download_permanent(self, file, powerloader_binary, checksums):
+        tag, name_on_server, username = oci_path_resolver(file, username="", name_on_server=file["name_on_mock_server"])
+        Path(get_oci_path(file, name_on_server, tag)[1]).unlink(missing_ok=True)
+        newpath, tmp_yaml = generate_oci_download_yml(file, tag, name_on_server, username, local=True)
+        download_oci_file(powerloader_binary, tmp_yaml, file, plain_http=True)
+        assert checksums[file["name_on_mock_server"]] == calculate_sha256(newpath)
+
+    # Upload a file and download it again
+    @pytest.mark.skipif(env_vars_absent() == False, reason="Environment variable(s) should not be defined")
     def test_upload_and_download(self, file, powerloader_binary):
-        username = self.set_username()
+        # Upload
         upload_path = generate_unique_file(file)
         hash_before_upload = calculate_sha256(upload_path)
-        tag, name_on_server = upload_oci(upload_path, powerloader_binary, file["oci_upload_location"])
+        tag, name_on_server = upload_oci(upload_path, powerloader_binary, file["oci_mock_server"], plain_http=True)
         Path(upload_path).unlink()
-        newpath, tmp_yaml = generate_oci_download_yml(file, tag, name_on_server, username)
-        download_oci_file(powerloader_binary, tmp_yaml, file)
+
```

Comment:
Can you check `api/v2/...` that it contains the tag and image?

Reply:
You can use `requests` library to make some simple HTTP requests.

Reply:
Yes, good idea :)

Reply:
Like this perhaps?
https://github.com/mamba-org/powerloader/blob/41dbc5109791b5689f7e09bea4cfb1455ebbafca/test/helpers.py#L144-L155

Reply:
yeah, does it work?